src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java

```diff
@@ -252,8 +252,7 @@ public DockerConfigFile getDockerConfig() {
 
     private AuthConfig getAuthConfig() {
         AuthConfig authConfig = null;
-        if (getRegistryUsername() != null && getRegistryPassword() != null && getRegistryEmail() != null
-                && getRegistryUrl() != null) {
+        if (getRegistryUsername() != null && getRegistryPassword() != null && getRegistryUrl() != null) {
```

Comment:
maybe better fill all non-null values into authconfig without combining them?

Reply:
something like
```
AuthConfig authConfig = new AuthConfig;
if (getRegistryUsername() != null) {
    authConfig.setRegistryUsername(getRegistryUsername());
}
if(getRegistryPassword() != nul) {
    authConfig.setRegistryPassword(getRegistryPassword());
}
...
````

Reply:
`Objects.nonNull()` and even in one liner

Reply:
Ok, but in `effectiveAuthConfig()` the return value of `getAuthConfig()` is checked for null. This will no longer work with your approach

Reply:
Then it will be empty, but not null...

Reply:
yes and because it is not null the empty config will be used for authetification instead of getting the auth config from the docker config file. So in `effectiveAuthConfig()` we need to check if the auth config returned by `getAuthConfig()` is empty. Which I think is ugly.

Maybe we should change the `&&` in `||`:
`if (getRegistryUsername() != null || getRegistryPassword() != null || getRegistryEmail() != null || getRegistryUrl() != null)`

@KostyaSha What di you think?

Reply:
well, if email optional, then ok as is